Update Substrate PalletXCM and Send XC-20s Overview (#1087)
moonbeam-foundation/moonbeam-docs: builders/interoperability/xcm/xc20/overview.md

From a technical standpoint, local XC-20s are ERC-20 tokens originating on Moonbeam (including bridged tokens deemed native once issued on Moonbeam), whereas external XC-20s are wrapped representations of tokens whose canonical ledger exists on another parachain or the relay chain. In all cases, XC-20s function just like standard ERC-20s—supporting common EVM-based use cases (such as DeFi, DEXs, and lending platforms)—but with the added advantage of seamless cross-chain operability.

![Moonbeam XC-20 XCM Integration With Polkadot](/images/builders/interoperability/xcm/overview/overview-3.webp)

This page aims to cover the basics on XC-20s; if you want to learn how to interact with or transfer XC-20s, please refer to the [Send XC-20s guide](/builders/interoperability/xcm/xc20/send-xc20s/overview/){target=\_blank}.

## Types of XC-20s {: #types-of-xc-20s }

There are two types of XC-20s: local and external.

> ### What are Local XC-20s? {: #local-xc-20s }

Reviewer: Lets use the same header tag, if we change them we might be leading external references to not finding the old section

Local XC-20s are all ERC-20s that exist on the EVM, and that can be transferred cross-chain through XCM. For local XC-20s to be transferred to another parachain, the asset must be registered on that chain. When transferring local XC-20s, the underlying tokens reside in the destination chain's Sovereign account on Moonbeam. A [sovereign account](/builders/interoperability/xcm/core-concepts/sovereign-accounts/){target=\_blank} is a keyless account governed by a blockchain runtime—rather than an individual—that can hold assets and interact with other chains. Local XC-20s must follow [the ERC-20 interface outlined in this guide](/builders/interoperability/xcm/xc20/interact/#the-erc20-interface){target=\_blank}. They must implement the standard ERC-20 function signatures, including the correct function selector of the `transfer` function as described in [EIP-20](https://eips.ethereum.org/EIPS/eip-20){target=\_blank}. However, additional functionality can still be added as long as it doesn’t break the base methods. 

> Note that, post-RT2301, creating a local XC-20 is equivalent to simply deploying a standard ERC-20 and enabling cross-chain features.

Reviewer: ```suggestion
Creating a local XC-20 is equivalent to deploying a standard ERC-20 and enabling cross-chain features on any Moonbeam network.
```

> ### What are External XC-20s? {: #external-xc-20s }

Reviewer: ```suggestion
### What are External XC-20s? {: #external-xc20s }
```

To keep it consistent with the previous header tag that existed

While Substrate assets are inherently interoperable, interacting with them requires using the Substrate API, which offers little visibility into the EVM. This creates a barrier for EVM-focused developers. To address this gap and extend Polkadot/Kusama’s native interoperability to the EVM, Moonbeam introduced the concept of XC-20s.

Initially, Moonbeam offered XC-20s as a specialized asset type, requiring cross-chain configuration during asset creation to enable full cross-chain capabilities. Now, since [runtime 2301](https://github.com/moonbeam-foundation/moonbeam/tree/runtime-2301){target=\_blank}, any ERC-20 token on Moonbeam can be configured for cross-chain transfers at any time effectively allowing all ERC-20s to function as XC-20s. Said differently, it is possible to make any ERC-20 an XC-20 at any time, by following the steps in the [XC Registration section](/builders/interoperability/xcm/xc-registration/){target=\_blank}.  

> XC-20s are a unique asset class on Moonbeam that blend the native interoperability of Substrate assets with a standard [ERC-20 interface](/builders/interoperability/xcm/xc20/interact/#the-erc20-interface){target=\_blank}. This means developers and users can interact with them just like any other ERC-20 token, with no Substrate knowledge needed, allowing seamless integration with EVM-based smart contracts (such as DEXs, lending platforms, and more) and familiar Ethereum development frameworks.

Reviewer 2: I think this definition of XC-20s should be before the backstory of them needing to be configured, but not anymore as of runtime 2301.

## Types of XC-20s {: #types-of-xc-20s }

There are two types of XC-20s: local and external.

### What are Local XC-20s? {: #local-xc20s }

Local XC-20s are all ERC-20s that exist on the EVM, and that can be transferred cross-chain through XCM. For local XC-20s to be transferred to another parachain, the asset must be registered on that chain. When transferring local XC-20s, the underlying tokens reside in the destination chain's Sovereign account on Moonbeam. Local XC-20s must follow [the ERC-20 interface outlined in this guide](/builders/interoperability/xcm/xc20/interact/#the-erc20-interface){target=\_blank}. They must implement the standard ERC-20 function signatures, including the correct function selector of the `transfer` function as described in [EIP-20](https://eips.ethereum.org/EIPS/eip-20){target=\_blank}. However, additional functionality can still be added as long as it doesn’t break the base methods. 

Note that, post-RT2301, creating a local XC-20 is equivalent to simply deploying a standard ERC-20 and enabling cross-chain features.

### What are External XC-20s? {: #external-xc20s }

> External XC-20s are native cross-chain assets originating from another parachain or the relay chain and have wrapped representations minted on Moonbeam. These assets are Substrate assets at their core. When transferring external XC-20s, the underlying native tokens reside in Moonbeam's Sovereign account in each chain. Recall that a Sovereign account is a special type of keyless account that can hold assets and interact with other chains and is governed by Moonbeam’s runtime rather than by any individual. The wrapped versions can then be freely utilized on Moonbeam.

Reviewer 2: Nice

> External XC-20s all have _xc_ prepended to their names to distinguish them as native cross-chain assets. For example, DOT, native to the Polkadot Relay Chain, is known as xcDOT when it exists in XC-20 form on Moonbeam. 

Reviewer 2: ```suggestion
External XC-20s all have _xc_ prepended to their names to distinguish them as native cross-chain assets. For example, DOT, native to the Polkadot relay chain, is known as xcDOT when it exists in XC-20 form on Moonbeam. 
```

### Local XC-20s vs External XC-20s {: #xc-20-comparison }

Both local and external XC-20s can be transferred to other parachains as if they were Substrate assets, using the Ethereum or Substrate API. However, when using the Substrate API, EVM logs are generated for local XC-20s but not for external ones. As a result, the Ethereum API is recommended for better visibility in EVM-based explorers such as [Moonscan](https://moonscan.io){target=\_blank}.

Within Moonbeam, local XC-20s can only be transferred via their standard ERC-20 interface. By contrast, external XC-20s support both the Substrate and ERC-20 interfaces. Remember that transfers of external XC-20s through the Substrate API will not appear in EVM-based explorers; only transactions performed via the Ethereum API are visible in those explorers.

> The key difference is that local XC-20s are EVM-native ERC-20s with XCM capabilities, whereas external XC-20s are inherently Substrate assets that expose an ERC-20 interface.

Reviewer 2: This is much better ✨

Cross-chain transfers of XC-20s are done using the Polkadot XCM Pallet. To learn how to use this pallet to transfer XC-20s, you can refer to the [Using the Polkadot XCM Pallet](/builders/interoperability/xcm/xc20/send-xc20s/xcm-pallet/){target=\_blank} guide.

> ## Asset Reserves 

Reviewer 2: This is first mentioning of asset reserves on this page, so we need to answer questions like what are asset reserves? What is the relevance when talking about XC-20s (since this is the XC-20 overview page)?

### Local Reserve Assets 

A local reserve asset on Moonbeam is a token whose canonical ledger—from an XCM perspective—resides natively on Moonbeam. In other words, Moonbeam is the asset’s home chain, where minting and burning take place. 

For example, Wormhole-wrapped ETH (wETH) is considered a local reserve asset on Moonbeam, even though Ethereum is the ultimate source of ETH. Once ETH is wrapped by Wormhole and enters the Polkadot ecosystem via Moonbeam, wETH can be transferred to other parachains through [Moonbeam Routed Liquidity (MRL)](/builders/interoperability/mrl/){target=\_blank}.

The important caveat is that, on a purely Ethereum-level view, ETH remains governed by and minted on Ethereum. However, from an XCM standpoint, wETH on Moonbeam is treated as a local reserve asset, meaning the canonical supply of wETH (as far as Polkadot ecosystems are concerned) exists on Moonbeam.

> ## Multilocation Overview

Reviewer 2: I'm not sure about having this section here if we have a whole page dedicated to it 🤷‍♀️

A [multilocation](/builders/interoperability/xcm/core-concepts/multilocations/){target=\_blank} is a structured way to pinpoint where an asset or account lives in the Polkadot/Kusama ecosystem. It specifies how many hops up or down the chain hierarchy are needed (the parents), and which detailed paths (the interior junctions) must be followed to get to a specific parachain, account, pallet, or asset.

When you see an XC-20 on Moonbeam that’s not native to Moonbeam (for example, xcDOT or xcASTR), it corresponds to a foreign asset whose reserve or home is on another parachain or the relay chain.

> Keep in mind that multilocations are always relative to a current position. For instance, defining a multilocation on Moonbeam differs from defining one on the Polkadot relay chain, even if both refer to the same asset. The multilocation identifies the asset’s origin chain and asset ID so the XCM virtual machine knows precisely where to lock, unlock, or update tokens. It identifies that asset’s origin chain and asset ID so that XCM knows:

Reviewer 2: ```suggestion
Keep in mind that multilocations are always relative to a current position. For instance, defining a multilocation on Moonbeam differs from defining one on the Polkadot relay chain, even if both refer to the same asset. The multilocation identifies the asset’s origin chain and asset ID so the Cross-Consensus Virtual Machine (XCVM) knows precisely where to lock, unlock, or update tokens. It identifies that asset’s origin chain and asset ID so that XCM knows:
```

I'm not sure about "It idefnities that asset's origin chain and asset ID so that XCM knows", specifically "XCM knows" sounds kinda weird to me haha 🤷‍♀️

From a technical standpoint, local XC-20s are ERC-20 tokens originating on Moonbeam (including bridged tokens deemed native once issued on Moonbeam), whereas external XC-20s are wrapped representations of tokens whose canonical ledger exists on another parachain or the relay chain. In all cases, XC-20s function just like standard ERC-20s—supporting common EVM-based use cases (such as DeFi, DEXs, and lending platforms)—but with the added advantage of seamless cross-chain operability.

![Moonbeam XC-20 XCM Integration With Polkadot](/images/builders/interoperability/xcm/overview/overview-3.webp)

This page aims to cover the basics on XC-20s; if you want to learn how to interact with or transfer XC-20s, please refer to the [Send XC-20s guide](/builders/interoperability/xcm/xc20/send-xc20s/overview/){target=\_blank}.

## Types of XC-20s {: #types-of-xc-20s }

There are two types of XC-20s: local and external.

> ### What are Local XC-20s? {: #what-are-local-xc-20s }

Reviewer 2: Did you by chance check to see if any links might reference #local-xc20s?

Reviewer 2: Same with #external-xc20s and any other headings that were modified

Local XC-20s are all ERC-20s that exist on the EVM, and that can be transferred cross-chain through XCM. For local XC-20s to be transferred to another parachain, the asset must be registered on that chain. When transferring local XC-20s, the underlying tokens reside in the destination chain's Sovereign account on Moonbeam. Local XC-20s must follow [the ERC-20 interface outlined in this guide](/builders/interoperability/xcm/xc20/interact/#the-erc20-interface){target=\_blank}. They must implement the standard ERC-20 function signatures, including the correct function selector of the `transfer` function as described in [EIP-20](https://eips.ethereum.org/EIPS/eip-20){target=\_blank}. However, additional functionality can still be added as long as it doesn’t break the base methods. 

Note that, post-RT2301, creating a local XC-20 is equivalent to simply deploying a standard ERC-20 and enabling cross-chain features.

### What are External XC-20s? {: #what-are-external-xc-20s }

> External XC-20s are cross-chain tokens originating from another parachain or the relay chain, and they are represented on Moonbeam as ERC-20 tokens. The original tokens remain locked in a sovereign account on their home chain, while the wrapped ERC-20 representation can be freely utilized on Moonbeam. A sovereign account is a keyless account governed by Moonbeam’s runtime—rather than an individual—that can hold assets and interact with other chains. When you transfer external XC-20s, the canonical assets remain in the sovereign account on their source chain, while the ERC-20 representation is what circulates on Moonbeam.

Author: Moved that sovereign account explainer and added a page on Sovereign accounts in the XCM Core Concepts Section.
